File: src/main/java/de/pschiessle/lumenforge/device/Device.java
package de.pschiessle.lumenforge.device;

import de.pschiessle.lumenforge.device.category.Category;
import de.pschiessle.lumenforge.device.maintenancestatus.MaintenanceStatus;
import de.pschiessle.lumenforge.device.vendor.Vendor;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.List;
import java.util.UUID;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Device {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @org.hibernate.annotations.UuidGenerator
    @Column(nullable = false, unique = true, updatable = false)
    private UUID uuid;

    @Column(name = "serial_number", unique = true)
    private String serialNumber;

    @Column(name = "device_name")
    private String name;

    @Column(name = "device_description")
    private String description;

    @Column(name = "photo_url")
    private String photoUrl;

    @ManyToOne(optional = false, fetch = FetchType.LAZY)
    @JoinColumn(name = "vendor_id", nullable = false)
    private Vendor vendor;

    @Column(name = "purchase_price", precision = 12, scale = 2)
    private BigDecimal purchasePrice;

    @Column(name = "purchase_date")
    private java.time.LocalDate purchaseDate;

    @ManyToMany
    @JoinTable(
            name = "device_category",
            joinColumns = @JoinColumn(name = "device_id"),
            inverseJoinColumns = @JoinColumn(name = "category_id")
    )
    private List<Category> categories;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "maintenance_status_id", nullable = false)
    private MaintenanceStatus maintenanceStatus;

    @OneToMany(mappedBy = "device", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<DeviceParameter> deviceParameters;

    @CreationTimestamp
    @Column(name = "create_date")
    private Instant createDate;

    @UpdateTimestamp
    @Column(name = "update_date")
    private Instant updateDate;
}

File: LumenForgeServer/LumenForgeServer/Inventory/Device.cs
using System;
using System.Collections.Generic;

namespace RentalDomain.Inventory;

public class Device
{
    public long Id { get; set; }
    public Guid Uuid { get; set; }
    public string SerialNumber { get; set; } = null!;
    public string? DeviceName { get; set; }
    public string? DeviceDescription { get; set; }
    public string? PhotoUrl { get; set; }

    public long VendorId { get; set; }
    public Vendor Vendor { get; set; } = null!;

    public decimal PurchasePrice { get; set; }
    public DateOnly PurchaseDate { get; set; }

    public long MaintenanceStatusId { get; set; }
    public MaintenanceStatus MaintenanceStatus { get; set; } = null!;

    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }

    public Stock Stock { get; set; } = null!;
    public List<DeviceParameter> Parameters { get; set; } = new();
    public List<DeviceCategory> DeviceCategories { get; set; } = new();
}
